esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -4286,21 +4308,33 @@ int CWsWorkunitsSoapBindingEx::onGetForm(IEspContext &context, CHttpRequest* req
                 request->getParameter("Range",range);
                 Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(cluster);
                 xml.append("<WUJobList>");
-                if (range.length())
-                    appendXMLTag(xml, "Range", range.str());
-                if (clusterInfo)
+
+                SecAccessFlags accessOwn;
+                SecAccessFlags accessOthers;
+                getUserWuAccessFlags(context, accessOwn, accessOthers, false);
+                if ((accessOwn == SecAccess_None) && (accessOthers == SecAccess_None))
+                {
+                    context.setAuthStatus(AUTH_STATUS_NOACCESS);
```

Comment:
Why is this set in getUserWuAccessFlags and here?  (same comment here and other places getUserWuAccessFlags is called)